fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -550,12 +582,28 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
 
         model.register(listener);
 
+        final AtomicBoolean propertyChanged = new AtomicBoolean();
+
+        ldpInsertedContentProperty(getNode()).ifPresent(resource -> {
+            final PropertyChangedListener propertyListener = new PropertyChangedListener(resource, propertyChanged);
+            model.register(propertyListener);
+        });
+
         model.setNsPrefixes(request.getPrefixMapping());
         execute(request, model);
 
         removeEmptyFragments();
 
         listener.assertNoExceptions();
+
+        try {
+            touch();
+            if (propertyChanged.get()) {
+                touchLdpMembershipResource(getNode());
+            }
+        } catch (final RepositoryException ex) {
+            throw new RepositoryRuntimeException(ex);
+        }
```

Comment:
Just a note, this `updateProperties()` method is getting denser and denser with no inline comments. This PR does not cause that problem, but it does add to it.